src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -914,6 +914,49 @@
     end
   end
 
+  describe 'POST #save_prjconf' do
+    before do
+      login user
+    end
+
+    context 'can save a project config' do
+      before do
+        post :save_prjconf, params: { project: user.home_project.name, config: 'save config' }
+      end
+
+      it { expect(flash[:success]).to eq('Config successfully saved!') }
+      it { expect(response.status).to eq(200) }
+    end
+
+    context 'cannot save a project config' do
+      before do
+        allow_any_instance_of(ProjectConfigFile).to receive(:save).and_return(nil)
+        post :save_prjconf, params: { project: user.home_project.name, config: '' }
+      end
+
+      it { expect(flash[:error]).not_to be_nil }
```

Comment:
@bgeuken, in this case, I preferred not test the message because I'm mocking `ProjectConfigFile`, so I would not have the full error message and it would be tested only part of the message.